### PR TITLE
Workflow invocation pagination

### PR DESCRIPTION
**Pull request by mvdbeek**
Adds pagination and sorting to the invocations component, and a pydantic model for all the endpoint parameters.

I have also added a `DecodedDatabaseIdField`, which if used in a Model, will provide a decoded id. I have another branch that builds on this and removes the bulk of the id de/encoding mess from the managers and other non-API code (https://github.com/galaxyproject/galaxy/compare/dev...mvdbeek:cleanup_id_decoding).

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 1]
```diff
@@ -846,14 +846,34 @@ def index_invocations(self, trans: GalaxyWebTransaction, workflow_id=None, **kwd
         else:
             # Get all invocation if user is admin
             user_id = target_user_id
-
+        sort_by = kwd.get('sortBy') or kwd.get('sort_by')
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/workflows.py]: Can we get away with just sort_by (and limit) or is there some Vue component that can't be configured to do that?

**Comment by mvdbeek** [comment]
That's ... not a lot to go on:
```
FAIL src/components/providers/monitors.test.js
  ● monitorHistoryUntilTrue › that history subscription runs until condition is met, and then once more

    connect ECONNREFUSED 127.0.0.1:80



  ● monitorHistoryUntilTrue › that history subscription runs until condition is met, and then once more

    connect ECONNREFUSED 127.0.0.1:80



  ● monitorHistoryUntilTrue › that history subscription runs until condition is met, and then once more

    connect ECONNREFUSED 127.0.0.1:80



  ● monitorHistoryUntilTrue › that history subscription runs until condition is met, and then once more

    connect ECONNREFUSED 127.0.0.1:80



```
https://github.com/galaxyproject/galaxy/runs/4858505190?check_suite_focus=true#step:7:386

**Comment by mvdbeek** [comment]
Alright, https://github.com/galaxyproject/galaxy/pull/13183/commits/b8bb2d4aaf38358171c7068771f5d25c1182e34a fixed it (on my fork at least ...) the trick is that one test function needs the module level jest.mock(), and the other doesn't and sometimes fails if it's there. So i've split them out. I guess overall not a bad idea to reduce mocking, although I don't get why that became a problem with the changes that I made in this PR

**Comment by davelopez** [comment]
xref https://github.com/galaxyproject/galaxy/pull/13435

**Pull request merged**